### PR TITLE
HOV-63 | Adds React Hook form example

### PR DESCRIPTION
**Pull request by pruthvi2103**
## Description

Adds a vite starter for react, adds hover, vanilla extract and implements reacthook form custom components

## Motivation and Context

Adds Example with React hook form for best practices using hover

## Screenshots (if any)

Screenshot 2023-01-20 at 3 55 48 PM

## How Has This Been Tested?

tested locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/with-react-hook-form/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Flex } from "@hover-design/react";
+import "./App.css";
+import Hero from "./components/Hero";
+
+import TextInput from "./components/form-components/text-input/TextInput";
+import SubmitButton from "./components/form-components/button-component/SubmitButton";
```
Comment (Raalzz) [inline examples/with-react-hook-form/src/App.tsx:6]: Can you create a `index.ts` file and export all the components from there? Imports will look good!

**Review by Raalzz — CHANGES_REQUESTED** [review]
comments

File: examples/with-react-hook-form/src/App.tsx (hunk 1 above)
Comment (pruthvi2103) [reply examples/with-react-hook-form/src/App.tsx:6]: on it

**Review by Raalzz — APPROVED** [review]
LGTM

**Pull request merged**